frontend/src/components/createUserModal/index.tsx

```diff
@@ -0,0 +1,194 @@
+import Button from '@material-ui/core/Button';
+import Clear from '@material-ui/icons/Clear';
+
+import { createStaff } from '../../features/staffSlice';
+import { toDateStr } from '../../constants/global';
+
+import 'toastr/build/toastr.min.css';
+import createUserModalStyle from './createUserModal.module.scss';
+
+interface Props {
+  closeModalFunction: () => void;
+}
+
+interface TextField {
+  id: string;
+  name: string;
+  type: string;
+}
+
+const textFields: TextField[] = [
+  { id: 'firstName', name: 'First Name', type: 'text' },
+  { id: 'surname', name: 'Surname', type: 'text' },
+  { id: 'jobTitle', name: 'Job title', type: 'text' },
+  { id: 'contractedHours', name: 'Contracted hours', type: 'number' },
+  { id: 'pay', name: 'Pay', type: 'number' },
+];
+
+const CreateUserModal = ({ closeModalFunction }: Props) => {
+  const dispatch = useDispatch();
+  const [preferredDays, setPreferredDays] = useState<{
+    [key: string]: boolean;
+  }>({
+    Mon: false,
+    Tue: false,
+    Wed: false,
+    Thu: false,
+    Fri: false,
+    Sat: false,
+    Sun: false,
+  });
+
+  const [userInfo, setUserInfo] = useState<{ [key: string]: string | number }>({
+    firstName: '',
+    surname: '',
+    jobTitle: '',
+    role: 'USER',
+    contractedHours: 0,
+    pay: 0,
+  });
+
+  const handlePreferredDaysChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setPreferredDays({
+      ...preferredDays,
+      [event.target.name]: event.target.checked,
+    });
+  };
+
+  const updateUserInfo = (userField: string, newValue: string | number) => {
+    setUserInfo({
+      ...userInfo,
+      [userField]: newValue,
+    });
+  };
+
+  const handleTextFieldChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const userField = event.target.name as string;
+    const newValueString = event.target.value as string;
+    let newValue: string | number;
+
+    if (event.target.type === 'number') {
+      newValue = parseFloat(newValueString);
+    } else {
+      newValue = newValueString;
+    }
+
+    updateUserInfo(userField, newValue);
+  };
+
+  const handleDropdownChange = (
+    event: React.ChangeEvent<{ name?: string | undefined; value: unknown }>
+  ) => {
+    const userField = event.target.name as string;
+    const newValue = event.target.value as string | number;
+    updateUserInfo(userField, newValue);
+  };
+
+  const handleCreateClick = () => {
+    const prefDaysString = Object.values(preferredDays)
+      .map((preferred) => +preferred)
+      .join('');
+    dispatch(
+      createStaff({
+        firstName: userInfo.firstName as string,
+        surname: userInfo.surname as string,
+        jobTitle: userInfo.jobTitle as string,
+        role: userInfo.role as 'USER' | 'MANAGER',
+        contractedHours: userInfo.contractedHours as number,
+        pay: userInfo.pay as number,
+        preferredDates: prefDaysString,
+        startDate: toDateStr(new Date()),
```

Comment:
Do we need `toDateStr` at all?
We now have `date-fns` as a dependency, so we can just import `formatISO` and use
`formatISO(new Date(), { representation: 'date' })`
https://date-fns.org/v2.13.0/docs/formatISO

Reply:
I did this because the way I defined the toDateStr function lets typescript know that what I am passing in is a string and at a date?